src/app/auth/components/signup/signup.component.ts

```diff
@@ -57,27 +60,28 @@ export class SignupComponent {
     if (
       this.signupForm.value.email &&
       this.signupForm.value.password &&
-      this.signupForm.value.username
+      this.signupForm.value.username &&
+      this.signupForm.value.role
     ) {
       const id = new Date().getTime().toString();
       const user: IUser = {
         email: this.signupForm.value.email,
         password: this.signupForm.value.password,
         username: this.signupForm.value.username,
+        role: this.signupForm.value.role,
```

Comment:
user is meant to have multiple roles, so it would be nice if you could change it to an array